patroni/api.py

```diff
@@ -175,13 +175,29 @@ def do_POST_failover(self):
         member = request.get('member', None)
         cluster = self.server.patroni.ha.dcs.get_cluster()
         status_code = 503
-        data = self.is_failover_possible(cluster, leader, member)
-        if not data:
-            if not self.server.patroni.dcs.manual_failover(leader, member):
-                data = b'failed to write failover key into DCS'
-            else:
-                self.server.patroni.dcs.event.set()
-                status_code, data = self.poll_failover_result(cluster.leader and cluster.leader.name, member)
+
+        data = b''
+        if request.get('planned_at'):
+            try:
+                planned_at = parse_datetime(request['planned_at'])
+                planned_at = localize_datetime(planned_at)
+                if planned_at < utcnow_timezone_aware():
+                    data = b'Cannot schedule failover in the past'
+                    status_code = 422
+                elif self.server.patroni.dcs.manual_failover(leader, member, planned_at):
+                    data = b'Failover scheduled'
+                    status_code = 200
+            except (ValueError, TypeError):
+                logger.exception('Invalid planned failover time: {}'.format(request['planned_at']))
+                data = b'Unable to parse planned timestamp. Should be in ISO 8601 format'
```

Comment:
True, we could relax this requirement. Instead of telling people it should be in this format we could mention that that is an unambiguous format that will always work, regardless of locale.